devel/prof_rc.txt

```diff
@@ -0,0 +1,209 @@
+
+Total time: 29.878 s
+File: /home/jaime/lib/ftprime/ftprime/argrecorder.py
+Function: add_record at line 39
+
+Line #      Hits         Time  Per Hit   % Time  Line Contents
+==============================================================
+    39                                               @profile
+    40                                               def add_record(self, left, right, parent, children):
+    41                                                   '''
+    42                                                   Add records corresponding to a reproduction event in which children (a
+    43                                                   tuple of IDs) inherit from parent (a single ID) on the interval
+    44                                                   [left,right).
+    45                                                   '''
+    46                                                   # unneeded but helpful for debugging
+    47    559608       799704      1.4      2.7          if parent not in self.keys():
+    48                                                       raise ValueError("Parent " + str(parent) +
+    49                                                                        "'s birth time has not been recorded with " +
+    50                                                                        ".add_individual().")
+    51                                                   # time = self[parent][0]
+    52    559608       339309      0.6      1.1          new_rec = msprime.Edgeset(
+    53    559608       276814      0.5      0.9                  parent=parent,
+    54    559608       256989      0.5      0.9                  children=children,
+    55    559608       254082      0.5      0.9                  left=left,
+    56    559608      1489470      2.7      5.0                  right=right)
+    57    559608     26461636     47.3     88.6          merge_records(new_rec, self[parent][1])
+
+Total time: 16.3576 s
+File: /home/jaime/lib/ftprime/ftprime/argrecorder.py
+Function: merge_records at line 142
+
+Line #      Hits         Time  Per Hit   % Time  Line Contents
+==============================================================
+   142                                           @profile
+   143                                           def merge_records(new, existing):
+   144                                               '''
+   145                                               Incorporate a new record (l,r,x,c,t[x])
+   146                                                 into a list of existing ones (a,b,x,C,t[x]) sorted on left endpoint.
+   147                                               Keeping them in sorted order simplifies the procedure
+   148                                                 (makes it so we don't have to split the new record).
+   149                                               '''
+   150    559608      1563370      2.8      9.6      k = 0
```

Comment:
huh, yes.  it would hit this without continuing to the rest of the function quite a bit, when we add offspring to new individuals (so len(existing) is zero).  maybe percentages are "average percent of the time spent on this line for each time the function is called" rather than "percent of total time spent in this function that was spent on this line"?